test/src/unittests/tonal/test_tonicindianartmusic.py

```diff
@@ -0,0 +1,119 @@
+# option) any later version.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+# details.
+#
+# You should have received a copy of the Afextentro GNU General Public License
+# version 3 along with this program. If not, see http://www.gnu.org/licenses/
+
+
+from essentia_test import *
+from numpy import sin, float32, pi, arange, mean, log2, floor, ceil, math, concatenate
+
+class TestTonicIndianArtMusic(TestCase):
+
+    def testInvalidParam(self):
+        self.assertConfigureFails(TonicIndianArtMusic(), { 'binResolution': -1 })
+        self.assertConfigureFails(TonicIndianArtMusic(), { 'frameSize': -1 })
+        self.assertConfigureFails(TonicIndianArtMusic(), { 'binResolution': 0 })
+        self.assertConfigureFails(TonicIndianArtMusic(), { 'frameSize': 0 })        
+        self.assertConfigureFails(TonicIndianArtMusic(), { 'harmonicWeight': -1 })
+        self.assertConfigureFails(TonicIndianArtMusic(), { 'harmonicWeight': 0 })
+        self.assertConfigureFails(TonicIndianArtMusic(), { 'harmonicWeight': 1 })
+        self.assertConfigureFails(TonicIndianArtMusic(), { 'hopSize': -1 })
+        self.assertConfigureFails(TonicIndianArtMusic(), { 'magnitudeCompression': -1 })
+        self.assertConfigureFails(TonicIndianArtMusic(), { 'magnitudeThreshold': -1 })
+        self.assertConfigureFails(TonicIndianArtMusic(), { 'maxTonicFrequency': -1 })
+        self.assertConfigureFails(TonicIndianArtMusic(), { 'minTonicFrequency': -1 })
+        self.assertConfigureFails(TonicIndianArtMusic(), { 'numberHarmonics': 0 })
+        self.assertConfigureFails(TonicIndianArtMusic(), { 'numberHarmonics': -1 })        
+        self.assertConfigureFails(TonicIndianArtMusic(), { 'numberSaliencePeaks': 0})
+        self.assertConfigureFails(TonicIndianArtMusic(), { 'referenceFrequency': -1 })
+        self.assertConfigureFails(TonicIndianArtMusic(), { 'sampleRate':   -1 })
+
+    def testZeros(self):
+        self.assertRaises(RuntimeError, lambda: TonicIndianArtMusic()(zeros(1024)))   
+
+    def testOnes(self):
+        referenceTonic =108.86099243164062     
+        tonic   = TonicIndianArtMusic()(ones(1024))
+        self.assertAlmostEqual(referenceTonic, tonic, 8)
+
+    def testNegativeInput(self):
+        tonic = TonicIndianArtMusic()([-1]*1024)
+        referenceTonic =108.86099243164062     
+        tonic   = TonicIndianArtMusic()([-1]*1024)
+        self.assertAlmostEqual(referenceTonic, tonic, 8)
+
+    def testRegression(self):
+        audio = MonoLoader(filename = join(testdata.audio_dir, 'recorded/vignesh.wav'),
+                            sampleRate = 44100)()
+        referenceTonic = 102.74                                       
+        tonic = TonicIndianArtMusic()(audio)
+        self.assertAlmostEqual(referenceTonic, tonic, 6)
```

Comment:
Same here, wrong order for test variables, and very large error margin. I wonder why is that. What is the output tonic `value`?